File: backend/src/adapters/mock.adapter.ts
import { ProviderAdapter, ProviderInvocationResult } from './types';

export class MockAdapter implements ProviderAdapter {
  async invoke(
    _credentials: Record<string, unknown>,
    model: string,
    payload: Record<string, unknown>
  ): Promise<ProviderInvocationResult> {
    return {
      output: {
        message: 'mock-response',
        model,
        payload
      },
      tokensIn: 50,
      tokensOut: 20,
      costUsd: 0
    };
  }
}

File: backend-java/src/main/java/com/neria/manager/adapters/MockAdapter.java
package com.neria.manager.adapters;

import java.util.Map;

public class MockAdapter implements ProviderAdapter {
  @Override
  public ProviderInvocationResult invoke(
      Map<String, Object> credentials, String model, Map<String, Object> payload) {
    Map<String, Object> output =
        Map.of(
            "id", "mock",
            "object", "chat.completion",
            "model", model,
            "choices", java.util.List.of(Map.of("message", Map.of("role", "assistant", "content", "Mock response"))),
            "usage", Map.of("prompt_tokens", 0, "completion_tokens", 0));
    return new ProviderInvocationResult(output, 0, 0, 0d);
  }
}
